erpnext/accounts/doctype/purchase_invoice/purchase_invoice.py

```diff
@@ -634,6 +634,29 @@ def make_item_gl_entries(self, gl_entries):
 								"cost_center": item.cost_center,
 								"project": item.project or self.project
 							}, account_currency, item=item))
+		
+						# check if the exchange rate has changed
+						purchase_receipt_conversion_rate = frappe.db.get_value('Purchase Receipt', {'name': item.purchase_receipt}, ['conversion_rate'])
+						if self.conversion_rate != purchase_receipt_conversion_rate:
+							gl_entries.append(
+								self.get_gl_dict({
+									"account": expense_account,
+									"against": self.supplier,
+									"debit": item.qty * item.rate * purchase_receipt_conversion_rate,
```

Comment:
@GangaManoj expense amount entry is already passed as per exchange rate in Purchase Invoice. Only difference amount should be handled here